### PR TITLE
Science controller

### PR DESCRIPTION
**Pull request by zameerbharwani**
With all the latest functionalities

### CODE REVIEW

File: Workspace/src/science_controller/src/science.py [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+                self.speed_dir = 0  # if still trying to move down do nothing
+
+        else:
+            self.speed_dir = joy.axes[1]
+            # if neither sensor is pressed,
+            # movement in either direction is allowed
+
+        shifter = rospy.get_param('budge')  # motor increment value
+
+        if self.temperature_switch == 1:
```
Comment (lijerryli) [inline Workspace/src/science_controller/src/science.py]: switch to constant velocity, just send -1 or 1 depending on button pressed, if nothing pressed then send 0

File: Workspace/src/science_controller/src/science.py [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+        if joy.buttons[3] == 1:  # X Button
+            self.which_sensor = 1  # EC sensor
+
+        elif joy.buttons[0] == 1:  # Y Button
+            self.which_sensor = 0  # Temp+Humidity sensor
+
+        if joy.buttons[2] == 1 or joy.buttons[1] == 1:
+            self.send_info_1()
+
+        if joy.axes[4] != 0:
```
Comment (lijerryli) [inline Workspace/src/science_controller/src/science.py]: always send the can messages associated with joypads (elevator + drill)

File: Workspace/src/science_controller/src/science.py [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+        if joy.axes[1] != 0:
+            self.send_info_3()
+
+        if joy.buttons[3] == 1 or joy.buttons[0] == 1:
+            self.send_info_4()
+
+        if joy.buttons[7] == 1 or joy.buttons[5] == 1:
+            self.send_info_5()
+
+    def send_info_1(self):
```
Comment (lijerryli) [inline Workspace/src/science_controller/src/science.py]: rename to more readable names

File: Workspace/src/science_controller/src/science.py [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+        self.pub.publish(canFrame5)
+
+    def send_flap(self):
+
+        frame = Frame()
+        frame.id = self.ID_flap
+        frame.is_rtr = False
+        frame.is_extended = False
+        frame.is_error = False
+        frame.dlc = 8
```
Comment (lijerryli) [inline Workspace/src/science_controller/src/science.py]: dlc = 4

File: Workspace/src/science_controller/src/science_no_LSs.py [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+and Y to read EC sensor. RT to move rack and pinion system up for
+temp probe motor. RB to move rack and pinion system down for temp
+probe motor.
+Author: Zameer Bharwani
+"""
+
+    def __init__(self):
+
+        self.init_Auger = rospy.init_node("ScienceController", anonymous=True)
+        self.sub_xbox = rospy.Subscriber("/joy", Joy, self.callback_xbox)
```
Comment (archielee) [inline Workspace/src/science_controller/src/science_no_LSs.py]: change to /science_joy

File: Workspace/src/science_no_LSs.py [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+and Y to read EC sensor. RT to move rack and pinion system up for
+temp probe motor. RB to move rack and pinion system down for temp
+probe motor.
+Author: Zameer Bharwani
+"""
+
+    def __init__(self):
+
+        self.init_Auger = rospy.init_node("ScienceController", anonymous=True)
+        self.sub_xbox = rospy.Subscriber("/joy", Joy, self.callback_xbox)
```
Comment (archielee) [inline Workspace/src/science_no_LSs.py]: change to science_joy

**Review by archielee — CHANGES_REQUESTED** [review]

**Pull request merged**